std/range/package.d

```diff
@@ -9213,3 +9221,297 @@ if (is(typeof(fun) == void) || isSomeFunction!fun)
 
     auto r = [1, 2, 3, 4].tee!func1.tee!func2;
 }
+
+/**
+Extend the length of the input range $(D r) by padding out the start of the
+range with the element $(D e). The element $(D e) must be of a common type with
+the element type of the range $(D r) as defined by $(D std.traits.CommonType).
+If $(D n) is less than the length of of $(D r), then the contents of $(D r) are
+returned.
+
+If $(D r) is an input range, this function is $(BIGOH 1). But if it is a forward
+range, it's $(BIGOH r.length).
+
+Params:
+    r = an input range with a length, or a string, or a forward range
+    e = element to pad the range with
+    n = the length to pad to
+
+Return:
+    a range containing the elements of the original range with the extra padding
+*/
+auto padLeft(R, E)(R r, E e, size_t n) if (
+    ((isInputRange!R && hasLength!R) || isNarrowString!R || isForwardRange!R) &&
+    !is(CommonType!(ElementType!R, E) == void))
+{
+    static if (hasLength!R || isNarrowString!R)
```

Comment:
I think narrow strings should go through the other branch. You want to pad for a specific minimum number of characters, not code units.